app/lib/actions/katello/repository/sync.rb

```diff
@@ -75,7 +79,7 @@ def humanized_name
         end
 
         def presenter
-          Helpers::Presenter::Delegated.new(self, planned_actions(Pulp::Repository::Sync))
+          Helpers::Presenter::Delegated.new(self, planned_actions(PulpSelector))
```

Comment:
did you try putting the code i mentioned above (https://github.com/Katello/katello/pull/8054/files#r273496245) here?  
 Instead of going through the pulp selector

Reply:
So what I noticed while trying a few things out is that inside the presenter method, we don't have access to any of the i/p or other variables from outside scope(or plan). So it's difficult to get the reposiotry to decide the action. 
Also, I tried skipping pulpselector and delegating directly to one of the orchestration actions but that does not seem to be supported right now. It requires to go one level down at a time. I can look at how we can change that?

Reply:
Ah!  I looked in the dynflow source, and it looks like we can use 'all_planned_actions' https://github.com/Dynflow/dynflow/blob/fdefa377dd52a9ae3e43ee241e874a1f77e84e32/lib/dynflow/action.rb#L222

```all_planned_actions(Pulp3::Repository::Sync)```

And this returns empty array if nothing is found, so i think we could just do something like:

```
found = all_planned_actions(Pulp::Repository::Sync)
found ||= all_planned_actions(Pulp3::Repository::Sync)
```